### PR TITLE
Improve model accuracy of fast training

### PR DESCRIPTION
**Pull request by Nic-Ma**
### Description
This PR updated the metrics algorithm in the fast training tutorial to train a more accurate model with slightly longer time.

### Status
**Ready**

### Checks

- [ ] Notebook runs automatically `./runner [-p <regex_pattern>]`

### CODE REVIEW

File: acceleration/fast_training_tutorial.ipynb [hunk 1]
```diff
@@ -18,9 +18,9 @@
     "3. Move data to GPU and cache, then execute random transforms on GPU.\n",
```
Comment (rijobro) [inline acceleration/fast_training_tutorial.ipynb:18]: <pre class="ql-syntax" spellcheck="false">Line #166.                        torch.save(model.state_dict(), "best_metric_model.pt")
</pre><p>I think we should be saving these models to a temporary dir and removing at the end. Otherwise we clutter the local repo with <code>.pt</code> files (even if they are ignored with <code>.gitignore</code>. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/687/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (Nic-Ma) [reply acceleration/fast_training_tutorial.ipynb:18]: That's a good point, I updated the PR for it.
Could you please help take a look again?

Thanks.

File: README.md [hunk 2]
```diff
@@ -175,7 +175,7 @@ And compares the training speed and memory usage with/without AMP.
 This notebook compares the performance of `Dataset`, `CacheDataset` and `PersistentDataset`. These classes differ in how data is stored (in memory or on disk), and at which moment transforms are applied.
 #### [fast_training_tutorial](./acceleration/fast_training_tutorial.ipynb)
 This tutorial compares the training performance of pure PyTorch program and optimized program in MONAI based on NVIDIA GPU device and latest CUDA library.
-The optimization methods mainly include: `AMP`, `CacheDataset` and `Novograd`.
+The optimization methods mainly include: `AMP`, `CacheDataset`, `GPU transforms`, `ThreadDataLoader`, `DiceCELoss` and `Novograd`.
```
[the remark this answers is not in the document]
Comment (Nic-Ma) [reply README.md]: Updated in the latest commit.

Thanks.

File: acceleration/fast_model_training_guide.md [hunk 3]
```diff
@@ -305,14 +305,13 @@ With all the above strategies, in this section, we introduce how to apply them t
...
   2. The deterministic transform results of all the spleen dataset is around 8 GB, which can be cached in a V100 GPU memory. So, we cached all the data in GPU memory and executed the following transforms in GPU directly.
 - Replace `DataLoader` with `ThreadDataLoader`. As all the data are cached in GPU, the computation of randomized transforms is on GPU and light-weighted, `ThreadDataLoader` help avoid the IPC cost of multi-processing in `DataLoader` and increase the GPU utilization.
 
-In summary, with a V100 GPU, we can achieve the training converges at a target validation mean Dice of `0.95` within one minute (`52s` on a V100 GPU, `41s` on an A100 GPU), it is approximately `200x` faster compared with the native PyTorch implementation when achieving the target metric. And each epoch is `20x` faster than the regular training.
+In summary, with a V100 GPU and the target validation `mean dice = 0.94` of the `forground` channel only,  it's more than `100x` speedup compared with the Pytorch regular implementation when achieving the same metric. And every epoch is `20x` faster than regular training.
```
[the remark this answers is not in the document]
Comment (Nic-Ma) [reply acceleration/fast_model_training_guide.md]: Updated in the latest commit.

Thanks.

**Review by dongyang0122 — APPROVED** [review]

**Comment by wyli** [comment]
Hi @Nic-Ma  is it feasible to provide a boolean flag `PROFILING` in this example? when `PROFILING=True`, it runs in a 'profiling mode' (using the nvtx utils to log more info, and running fewer total epochs https://github.com/Project-MONAI/MONAI/blob/dev/monai/utils/nvtx.py) cc @AHarouni @dongyang0122
